web/src/frontend/src/nav-soknad/components/timeoutbox/TimeoutBox.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import NavFrontendModal from "nav-frontend-modal";
+import Nedtelling from "./Nedtelling";
+import LoggetUt from "./LoggetUt";
+import "./timeoutbox.css";
+
+interface State {
+	sesjonGarUt: number;
+	visAdvarsel: number;
+	apen: boolean;
+	sjekkGjenstaendeTid: any;
+}
+
+interface Props {
+	sessionDurationInMinutes: number;
+	showWarningerAfterMinutes: number;
+}
+
+const now = () => {
+	return new Date().getTime();
+};
+
+class TimeoutBox extends React.Component<Props, State> {
+
+	constructor(props: Props) {
+		super(props);
+		this.state = {
+			sesjonGarUt: now() + (this.props.sessionDurationInMinutes * 60 * 1000),
+			visAdvarsel: now() + (this.props.showWarningerAfterMinutes * 60 * 1000),
+			apen: false,
+			sjekkGjenstaendeTid: null
+		};
+	}
+
+	componentWillMount() {
+		const sjekkGjenstaendeTid = setInterval(() => {
+			const tidIgjenAvSesjon = this.state.sesjonGarUt - now();
+			const tidIgjenForAdvarsel = this.state.visAdvarsel - now();
+			const apen = (tidIgjenAvSesjon < 0) || (tidIgjenForAdvarsel < 0);
+			this.setState({
+				apen
+			});
+		}, 1000);
+		this.setState({sjekkGjenstaendeTid});
+	}
+
+	componentWillUnmount() {
+		clearInterval(this.state.sjekkGjenstaendeTid);
+	}
+
+	onCloseModal() {
+		this.setState({ apen: false });
+	}
+	onLoginAgainClick() {
+		window.location.reload();
+	}
+
+	onContinueClick() {
+		this.setState({
+			sesjonGarUt: now() + (this.props.sessionDurationInMinutes * 60 * 1000),
+			apen: false
+		});
+	}
+
+	render() {
+		const visAdvarsel = ((this.state.visAdvarsel - now()) < 0);
+		return (
+			<NavFrontendModal
+				isOpen={this.state.apen}
+				contentLabel="Fortsett"
+				onRequestClose={() => {
+					this.onCloseModal();
+				}}
+			>
+				{visAdvarsel && (
```

Comment:
nedtelling og loggetUt ser veldig like ut - kunne man laget en felleskomponent og så heller endret teksten man sendte inn og lenke?

Reply:
I utgangspunktet helt enig. Etter at jeg la inn react-intl, så ble Nedtelling.tsx litt mer komplisert, og egentlig helt greit å ha i egen komponent. Det eneste komponentene har felles er dette, så jeg tror vi kan ha det som to komponenter.

```
			<Innholdstittel className="blokk-s">
				<FormattedMessage id={"timeout.overskrift"}/>
			</Innholdstittel>
```

Reply:
I utgangspunktet helt enig. Etter at jeg la inn react-intl, så ble Nedtelling.tsx litt mer komplisert, og egentlig helt greit å ha i egen komponent. Det eneste komponentene har felles er dette, så jeg tror vi kan ha det som separate komponenter.

```
			<Innholdstittel className="blokk-s">
				<FormattedMessage id={"timeout.overskrift"}/>
			</Innholdstittel>
```

Reply:
Kikket kanskje litt raskt over, men vil ikke denne meldingen komme opp etter 25 minutter nå, uansett om bruker har vært aktiv eller ikke? Dersom bruker gjør noe mot server, så skal vi vel starte timeren på nytt?